hcl2template/addrs/plugin.go

```diff
@@ -13,21 +17,53 @@ import (
 
 // Plugin encapsulates a single plugin type.
 type Plugin struct {
-	Hostname  string
-	Namespace string
-	Type      string
+	Source string
 }
 
-func (p Plugin) RealRelativePath() string {
-	return p.Namespace + "/packer-plugin-" + p.Type
+// Parts returns the list of components of the source URL, starting with the
+// host, and ending with the name of the plugin (non-simplified).
```

Comment:
Oh that might be out-of-scope given that I removed the `packer-plugin-<name>` support from the URIs after we discussed it out-of-band.

The `non-simplified` part here meant that the name of the plugin might be in the `packer-plugin-amazon` form given that I had removed the check during installation, but since it's been reverted, the `name` should always be `packer-plugin-` free now.